### DISCUSSION TITLE
The `COLLATION_CHARACTER_SET_APPLICABILITY` table seems to be missing

### DISCUSSION THREAD

**Discussion by billy1624** [planetscale/discussion#155] [Bug Reports]
Hi, I got a database error when query some table info on Planetscale.

The raw SQL:
```sql
SELECT
    `table_name`,
    `engine`,
    `auto_increment`,
    `table_collation`,
    `table_comment`,
    `create_options`,
    `collation_character_set_applicability`.`character_set_name`
FROM
    `information_schema`.`tables`
    LEFT JOIN `information_schema`.`collation_character_set_applicability` ON `collation_character_set_applicability`.`collation_name` = `tables`.`table_collation`
WHERE
    `table_schema` = ?
    AND `table_type` = ?
ORDER BY
    `table_name` ASC
```

This error was thrown out of [SeaORM](https://github.com/SeaQL/sea-orm) CLI:
```
value: Database(MySqlDatabaseError { code: Some("HY000"), number: 1105, message: "symbol collation_character_set_applicability.collation_name not found" })
```

I guess although Planetscale compatible with MySQL but it does not support all features provided by MySQL?
- More info on the INFORMATION_SCHEMA COLLATION_CHARACTER_SET_APPLICABILITY Table: https://dev.mysql.com/doc/refman/8.0/en/information-schema-collation-character-set-applicability-table.html

Thanks!

**Comment by jgreet** [comment]
Something is definitely wrong here, but that table should be accessible:

```sql
$ pscale shell vdemo main
vdemo/|⚠ main ⚠|> select count(*) from information_schema.collation_character_set_applicability\G
*************************** 1. row ***************************
count(*): 272
```

This might be an issue with the Vitess query planner. 

I'm not familiar with SeaORM, are you seeing that error running something like this?
```
sea-orm-cli generate entity --database-url mysql://username:pscale_pw_XXXXXX@branchid.us-east-2.psdb.cloud/db-name
```

**Reply by billy1624** [reply]
> ```sql
> $ pscale shell vdemo main
> vdemo/|⚠ main ⚠|> select count(*) from information_schema.collation_character_set_applicability\G
> *************************** 1. row ***************************
> count(*): 272
> ```

This works but below query failed

```sql
SELECT
    `table_name`,
    `engine`,
    `auto_increment`,
    `table_collation`,
    `table_comment`,
    `create_options`,
    `collation_character_set_applicability`.`character_set_name`
FROM
    `information_schema`.`tables`
    LEFT JOIN `information_schema`.`collation_character_set_applicability` ON `collation_character_set_applicability`.`collation_name` = `tables`.`table_collation`
WHERE
    `table_schema` = 'xxx'
    AND `table_type` = 'xxx'
ORDER BY
    `table_name` ASC
```

```
[HY000][1105] symbol collation_character_set_applicability.collation_name not found
```

**Reply by jgreet** [reply]
The query above succeeds with the upcoming `Gen4` query planner enabled in Vitess.

Similar to the [Metabase issue](https://github.com/planetscale/beta/discussions/161#discussioncomment-2161079), we could flag individual databases in to the new planner but I suspect you're likely to run in to other compatibility issues.

Testing against a database set to use the Gen4 query planner:

```
sea-orm-cli generate entity --database-url mysql://username:pscale_pw_xxxx@xxxx.us-east-1.psdb.cloud/dbname

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ColumnDecode { index: "0", source: "mismatched types; Rust type `alloc::string::String` (as SQL type `VARCHAR`) is not compatible with SQL type `VARCHAR`" }', /Users/jgreet/.cargo/registry/src/github.com-1ecc6299db9ec823/sqlx-core-0.5.10/src/row.rs:73:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Perhaps that's related to https://github.com/planetscale/beta/discussions/162

**Comment by deepthi** [comment]
Can you try the query from the CLI with the fully-qualified column name?
```
LEFT JOIN `information_schema`.`collation_character_set_applicability` ON `information_schema`.`collation_character_set_applicability`.`collation_name` = `tables`.`table_collation`
```
